core/dbt/clients/system.py

```diff
@@ -61,11 +61,13 @@ def find_matching(
                 relative_path = os.path.relpath(
                     absolute_path, absolute_path_to_search
                 )
+                modification_time = os.path.getmtime(absolute_path)
```

Comment:
It looks like `getmtime` can raise an `OSError`. We should probably explicitly handle that case.

Reply:
It might be a good idea, but this code is all being reimplemented in the filesystem adapter, so this is going to be temporary. Also we just got the list of files from os.path, so there's not much of a chance that the file isn't there.

Reply:
I agree that it's unlikely to come up, but I think it also throws if the metadata doesn't exist, or is inaccessible independent of the file existing. Given that dbt runs on a ton of machines, I'd like to handle all our edge cases if possible especially since this is an exception we can eat. If the last updated timestamp isn't there, just reload it like normal.

Reply:
There won't be any thrown exceptions. I checked that getmtime works correctly on Windows. And it certainly works on Mac and Linux. We don't support any other platforms. The test suite has been run on all of our integration environments and it works correctly.

That file just came from os.walk. So if it doesn't exist it could only be from a bug in os.walk, which is unlikely and would have all kinds of other implications. The code to get those files has been there since the beginning and if there was a possibility of the file not being there we would have seen errors before.

I agree that exceptions should be caught when they're possible. This is not a place where they're possible. Adding an unnecessary exception complicates the code and is overly defensive.

Reply:
I land right in the middle, it shouldn't happen but if it does, let's raise something a _little_ more helpful maybe?
```suggestion
				try:
                	modification_time = os.path.getmtime(absolute_path)
                except OSError as exc:
                	# this shouldn't be reachable, raise something a little more helpful
                	raise dbt.exceptions.InternalException(
        				f'Unhandled exception getting modified time for file "{absolute_path}": {exc}'
    				) from exc

```

Reply:
`getmtime` is dependent on the filesystem not the OS and we definitely do not test across all the filesystems our users are on. Hence, my strong preference to code defensively here.

@kwigley if we're going to catch it, I feel like we should just eat the exception by passing None. If the modification time doesn't exist, then dbt would fallback to the original behavior of reloading. But I will agree, throwing your suggested exception at least lets us debug it if it happens, but it would still happen.

Reply:
Ah, ok. That's a distinction I didn't grok in my first pass, I thought that this would only raise if something went really bad or some race condition where we read a file and the file disappeared when trying to get the modified time. I don't have a sense of how often this could raise... but catching, logging and moving along seems appropriate. Though, I'm sure something will throw elsewhere :)
```suggestion
                modification_time = time.time() # <- I don't think this can be None?
                try:
                    modification_time = os.path.getmtime(absolute_path)
                except OSError as exc:
                    logger.exception('uh oh')
```

Reply:
I think `None` makes more sense though